sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2696,7 +2699,13 @@ case class Encode(value: Expression, charset: Expression)
   protected override def nullSafeEval(input1: Any, input2: Any): Any = {
     val toCharset = input2.asInstanceOf[UTF8String].toString
     try {
-      input1.asInstanceOf[UTF8String].toString.getBytes(toCharset)
+      if (legacyCharsets ||
+        toCharset.equalsIgnoreCase("UTF-8") || toCharset.equalsIgnoreCase("US-ASCII") ||
+        toCharset.equalsIgnoreCase("ISO-8859-1") ||
+        toCharset.equalsIgnoreCase("UTF-16") || toCharset.equalsIgnoreCase("UTF-16LE") ||
+        toCharset.equalsIgnoreCase("UTF-16BE")) {
```

Comment:
Shall we add `val supportedCharsets = Set("UTF-8", ...)`? So we can simplify these code.

Reply:
Added